evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -73,7 +73,8 @@ def next_batch(self):
                 pipeline_class = pipeline_dict['pipeline_class']
                 pipeline_params = pipeline_dict['parameters']
                 input_pipelines.append(pipeline_class(parameters=self._transform_parameters(pipeline_class, pipeline_params)))
-            ensemble = _make_stacked_ensemble_pipeline(input_pipelines, input_pipelines[0].problem_type)
+            ensemble = _make_stacked_ensemble_pipeline(input_pipelines, input_pipelines[0].problem_type,
+                                                       random_state=self.random_state)
```

Comment:
Ooo good catch indeed. @dsherry the [sklearn stacked ensembler](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.StackingClassifier.html#sklearn.ensemble.StackingClassifier) doesn't accept `random_state` as a parameter, so its enforced by the passed input estimators. So if `random_state` is passed there then yes!

Reply:
Great catch guys! The StackedEnsembles don't overwrite the random state of the input pipelines. The random state is only used for the CV. I added unit tests to document that behavior.

Since we want the pipelines in the stacked ensemble to have the random state of the `IterativeAlgorithm` - I now pass the random state to each pipeline as they are created (before `_make_stacked_ensemble_pipeline`).